### ISSUE TITLE
Zone's 2+ don't calculate new time and don't start after one sucefull complete run. Zone 1 is working fine, but zone's 2+ don't showing start times. After a manually deactivation of the zone, they show the start time. 

### ISSUE THREAD

**Issue by dAjaY85** [rgc99/irrigation_unlimited#67]
Home assistant version 2022.4.7
Irrigation Unlimited: 2022.4.0

**Comment by rgc99** [comment]
Please post your configuration and log entries.

**Comment by dAjaY85** [comment]
My Config is:
`
irrigation_unlimited:
  controllers:
  - name: "Sonoff_4CH_Pro_Bewaesserung"
    zones:
      - entity_id: "switch.bewaeserung_rasen_seitlich"
      - entity_id: "switch.bewaeserung_rasen_hinten"
      - entity_id: "switch.bewaeserung_beete_hinten"
    sequences:
      - name: "Summer"
        duration: "00:15"
        delay: "00:01"
        schedules:
          - name: "Sunrise"
            time:
              sun: "sunrise"
          - name: "After sunset"
            time:
              sun: "sunset"
              after: "00:30"
            month: [may, jun, jul, aug]
        zones:
          - zone_id: 1
          - zone_id: 2
          - zone_id: 3
            duration: "00:40"
      - name: "Spring and Autumn"
        duration: "00:10"
        delay: "00:01"
        schedules:
          - name: "Sunrise"
            time:
              sun: "sunrise"
          - name: "After sunset"
            time:
              sun: "sunset"
              after: "00:30"
            month: [mar, apr, sep, oct, nov]
        zones:
          - zone_id: 1
          - zone_id: 2
          - zone_id: 3
            duration: "00:40"   
`

**Comment by rocheuf** [comment]
Hello,
Same behavior for me : zone 2 does not start anymore. Everything worked fine last year with no change in my setup.

my config :

irrigation_unlimited:
  controllers:
    zones:
      - name: ‘devant’
        entity_id: 'switch.arrosage1_switch'
      - name: ‘derriere’
        entity_id: 'switch.arrosage2_switch'
    sequences:
      - duration: '00:05'
        delay: '00:01'
        schedules:
          - time: '05:00'
            day: [1, 4, 7, 10, 13, 16, 19, 22, 25, 28]
            month: [apr, may, jun, jul, aug, sep, oct]
        zones:
          - zone_id: 1
          - zone_id: 2

**Comment by rgc99** [comment]
It appears to work as expected. To go further I need the log entries in particular the service calls.

Not sure if you realise that zone 2 overlaps with zone 1 and 3. The "Sunrise" schedules in each sequence are the same so both sequences will start together. Did you mean to put a month qualifier in them like you have in the "After sunset" schedules?

**Comment by pattisonmichael** [comment]
I have a similar problem, but it's opposite. I have 3 zones in 1 sequence. After reloading the config each zones show as getting equal time to run, but after the first run, Zone 1 will not be scheduled anymore. A restart of HA doesn't fix it, only reloading the config will solve it after a few minutes.

```
  controllers:
    name: "Garden Water"
    enabled: true
    preamble: "00:05"
    all_zones_config: # <= Add these three lines <─┐
      show: # <= to the configuration            <─┤
        timeline: true # <= for all zones        <─┘
      maximum: "01:00"
    zones:
      - name: "Plants"
        entity_id: "switch.garden_sprinkler_1"
      - name: "Lawn"
        entity_id: "switch.garden_sprinkler_2"
      - name: "Side"
        entity_id: "switch.garden_sprinkler_3"
    sequences:
      - name: "Default Sequence"
        repeat: 2
        schedules:
          - name: "Summer"
            duration: "02:00"         
            time: "19:00"           
        zones:
          - zone_id: 1
          - zone_id: 2
          - zone_id: 3
```
![image](https://user-images.githubusercontent.com/17926363/168889671-1eb6a848-173a-4fe5-9f3c-51bda479678e.png)

**Comment by rgc99** [comment]
I need the log entries. It seems now in HA in Settings -> System -> Logs -> 'Load Full Logs' now only show the last 50 entries. Can't seem to change this so it scrolls out of view. The data is there in home-assistant.log but I don't know how to easily tell someone to get it. If you understand SSH then I need the lines where custom_components.irrigation_unlimited appears in. i.e. `grep custom_components.irrigation_unlimited home-assistant.log` or install samba, mount the config folder and use a text editor.

Not sure what I am missing here. Log files are not much good if you can't get to them easily.

**Comment by pattisonmichael** [comment]
I got this from the log, what I found strange is that there is a zone 0,1,2,3 but I only have 3 zones in total...
```
2022-05-17 16:26:27 INFO (MainThread) [custom_components.irrigation_unlimited] 
2022-05-17 18:55:04 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 18:55:00] controller: 1, zone: 0, state: 1
2022-05-17 19:00:04 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 19:00:00] controller: 1, zone: 2, state: 1, data: 2.1.1.2.1
2022-05-17 19:20:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 19:20:00] controller: 1, zone: 2, state: 0
2022-05-17 19:20:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 19:20:00] controller: 1, zone: 3, state: 1, data: 3.1.1.3.1
2022-05-17 19:40:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 19:40:00] controller: 1, zone: 3, state: 0
2022-05-17 19:40:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 19:40:00] controller: 1, zone: 2, state: 1, data: 2.1.1.2.1
2022-05-17 20:00:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 20:00:00] controller: 1, zone: 2, state: 0
2022-05-17 20:00:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 20:00:00] controller: 1, zone: 3, state: 1, data: 3.1.1.3.1
2022-05-17 20:20:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 20:20:00] controller: 1, zone: 3, state: 0
2022-05-17 20:20:05 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 20:20:00] controller: 1, zone: 0, state: 0
2022-05-17 20:45:07 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 20:45:00] controller: 1, zone: 0, state: 1
2022-05-17 20:50:08 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 20:50:00] controller: 1, zone: 1, state: 1, data: 1.0.0.0.0
2022-05-17 21:30:08 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 21:30:00] controller: 1, zone: 1, state: 0
2022-05-17 21:30:08 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-17 21:30:00] controller: 1, zone: 0, state: 0

```

**Comment by pattisonmichael** [comment]
I think i captured it in the log now, bit still don't understand why...
It is caused by the automation "Irrigation Unlimited Adjustment".
```
2022-05-18 12:37:52 INFO (MainThread) [custom_components.irrigation_unlimited]
2022-05-18 12:37:52 DEBUG (MainThread) [custom_components.irrigation_unlimited] LOAD
2022-05-18 12:37:54 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-05-18 12:37:00] controller: 0, zone: 0, entity: irrigation_unlimited.coordinator
2022-05-18 12:38:01 DEBUG (MainThread) [custom_components.irrigation_unlimited] START
2022-05-18 12:38:07 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-05-18 12:38:00] controller: 1, zone: 0, entity: binary_sensor.irrigation_unlimited_c1_m
2022-05-18 12:38:07 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-05-18 12:38:00] controller: 1, zone: 1, entity: binary_sensor.irrigation_unlimited_c1_z1
2022-05-18 12:38:07 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-05-18 12:38:00] controller: 1, zone: 2, entity: binary_sensor.irrigation_unlimited_c1_z2
2022-05-18 12:38:07 DEBUG (MainThread) [custom_components.irrigation_unlimited] REGISTER [2022-05-18 12:38:00] controller: 1, zone: 3, entity: binary_sensor.irrigation_unlimited_c1_z3
2022-05-18 12:39:35 DEBUG (MainThread) [custom_components.irrigation_unlimited] INITIALISED
2022-05-18 18:55:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 18:55:00] controller: 1, zone: 0, state: 1
2022-05-18 19:00:20 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 19:00:00] controller: 1, zone: 1, state: 1, data: 1.1.1.1.1
2022-05-18 19:20:20 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 19:20:00] controller: 1, zone: 1, state: 0
2022-05-18 19:20:20 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 19:20:00] controller: 1, zone: 2, state: 1, data: 2.1.1.2.1
2022-05-18 19:40:20 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 19:40:00] controller: 1, zone: 2, state: 0
2022-05-18 19:40:20 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 19:40:00] controller: 1, zone: 3, state: 1, data: 3.1.1.3.1
2022-05-18 20:00:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 20:00:00] controller: 1, zone: 3, state: 0
2022-05-18 20:00:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 20:00:00] controller: 1, zone: 1, state: 1, data: 1.1.1.1.1
2022-05-18 20:20:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 20:20:00] controller: 1, zone: 1, state: 0
2022-05-18 20:20:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 20:20:00] controller: 1, zone: 2, state: 1, data: 2.1.1.2.1
2022-05-18 20:40:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 20:40:00] controller: 1, zone: 2, state: 0
2022-05-18 20:40:21 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 20:40:00] controller: 1, zone: 3, state: 1, data: 3.1.1.3.1
2022-05-18 21:00:22 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 21:00:00] controller: 1, zone: 3, state: 0
2022-05-18 21:00:22 INFO (MainThread) [custom_components.irrigation_unlimited] EVENT [2022-05-18 21:00:00] controller: 1, zone: 0, state: 0
2022-05-18 23:30:00 DEBUG (MainThread) [custom_components.irrigation_unlimited] CALL [2022-05-18 23:30:00] service: adjust_time, controller: 1, zone: 1, data: {"entity_id": "binary_sensor.irrigation_unlimited_c1_z1", "actual": "0:00:00"}

```

**Comment by rgc99** [comment]
Perfect, just what I needed. Zone 1 is being nailed by a service call. I am guessing this might be trying to use HAsmartirrigation and not working properly. Anyway, post your automation so we can sort it out.

**Comment by pattisonmichael** [comment]
It is the automation provided by you :) I haven't setup a new one and I'm not using HASmartIrrigation yet.
Can I just disable that automation?

**Comment by rgc99** [comment]
Yes, disable it. The script needs to have a condition to deal with this situation that HAsmartirrigation is not there.

You will need to reset the adjustment on zone 1. A restart won't do it as the adjustments will persist. Go into Developer Tools -> Services. Click on 'GO TO YAML MODE'. Paste this and press 'CALL SERVICE'

```yaml
service: irrigation_unlimited.adjust_time
data:
  entity_id: binary_sensor.irrigation_unlimited_c1_z1
  reset: null
```

**Comment by rgc99** [comment]
The [automation](https://github.com/rgc99/irrigation_unlimited/blob/master/packages/irrigation_unlimited_smart_irrigation.yaml) has been updated and now checks HAsmartirrigation is installed. You will have to configure the automation for the correct zones/sequences before it will operate. It is not simply an out of the box solution as each system setup is different. Please refer to the comments in the file.

**Comment by pattisonmichael** [comment]
That makes sense to me and after disabling the automation my problem was solved.
I like your approach to check for the dependency first!
But in the new version of the automation wouldn't the second automation "Smart Irrigation reset bucket" now throw an error after every run of the first controller if HAsmartirrgation isn't installed (because the service won't be there)?
I might give it a try tonight and report back.

**Comment by rgc99** [comment]
You are correct about the second automation although it won't have any side effect apart from the error message, unlike the the first. I have added the same conditions to this automation as well.

**Comment by rgc99** [comment]
I have posted a better solution that checks everything is setup.

**Comment by dAjaY85** [comment]
Today updated to the new version, seems working with this one better. Will test some days.

Thank you for this version.

**Comment by hillbicks** [comment]
Thank you for the fix. Was already pulling my hair out, seems like it is fixed now.

**Comment by dAjaY85** [comment]
Looks like it's fixed now, I close this topic.

Thank you for this fix.
